design/010-new-dataset-fields/010-new-dataset-fields.md

```diff
@@ -0,0 +1,158 @@
+- Start Date: 22-07-2022
+- Status: Proposed
+
+# New metadata fields for datasets
+
+## Summary
+
+Add standard metadata fields to make Renku datasets more FAIR.
+
+## Motivation
+
+As a Renku user, I want to easily annotate my datasets with standard metadata. Renku datasets support custom annotations as JSON-ld files. Although these annotations are available in the project graph, they are not in the global KG. Furthermore, the process of adding custom metadata is tedious and error prone (requires writing JSON-ld by hand). Currently, some crucial default metadata fields (e.g. license) are missing from datasets.
+
+Datasets metadata should support common fields required for FAIR-ification.
+
+## Design Detail
+
+### User research data
+
+We can leverage information from 3 sources to identify potential fields. In order of decreasing priority: FAIR recommendations, User feedback, User persona.
+
+**FAIR recommendations**
+
+* F1: (Meta)data are assigned a globally unique and persistent identifier
+  + _Ideally we would use something like ROR for **affiliation**_
+* R1.1: data are released with a clear and accessible data usage **license**
+
+**Feedback**
+
+* I would like to know the total dataset **size** to adjust session ressources and avoid running out of disk space
+* Dataset imports are slow
+  + _A **size** attribute may help by showing an ETA or even “large dataset” warning in the future._
+* I would like to search entities by **affiliation** and **topic** in the search page
+
+**Persona**
+
+* No visibility on my project
+  + _Unique **affiliation** (ROR, ORCID) + **topic** may help with that_
+
+### Proposed fields
+
+The following fields can be identified from the data above. They are listed in order of priority with the following logic: FAIR recommendation > user feedback > user persona.
+
+**License**
+
+Data usage license. The license cannot live _only_ as metadata and there must also be a file in the project. Therefore, this field should be populated when the file is added. In the future, Renku could propose a (graphical/no code) way to choose a license, which would both populate the metadata field and add the file. Here, we simply propose to make this optional field available for future use.
```

Comment:
Datasets already have a `license` field: https://github.com/SwissDataScienceCenter/renku-python/blob/fb76eadb98727589fee51f2d4df768f475a704c1/renku/command/schema/dataset.py#L163. It's populated when importing a dataset. ATM, it's not possible to set/edit this field via CLI/UI. Moreover, it only accepts a URL but according to schema.org (https://schema.org/license) it should be possible to have a `schema:CreativeWork` that should address the mentioned file use-case.

Reply:
How do you think `CreativeWork` would support the file use-case ? Do you mean by allowing to easily create a file from the CreativeWork element ?

If so, SPDX also provides JSON with the license contents for each license URL, e.g. https://spdx.org/licenses/BSD-3-Clause.json

Do you think that would work ?

Reply:
I guess either we can have a file as mentioned in the use-case by using `schema:DigitalDocument` which is derivative from `schema:CreativeWork` or we can include the license text in the metadata directly (and not having a file) by using https://schema.org/text.

Reply:
I added the alternative solution with DigitalDocument to the document !